updater/fetchers/alpine/alpine.go

```diff
@@ -0,0 +1,247 @@
+	repositoryLocalPath string
+}
+
+func (f *fetcher) FetchUpdate(db database.Datastore) (resp updater.FetcherResponse, err error) {
+	log.Info("fetching Alpine vulnerabilities")
+
+	// Pull the master branch.
+	var commit string
+	commit, err = f.pullRepository()
+	if err != nil {
+		return
+	}
+
+	// Ask the database for the latest commit we successfully applied.
+	var dbCommit string
+	dbCommit, err = db.GetKeyValue(updaterFlag)
+	if err != nil {
+		return
+	}
+
+	// Set the updaterFlag to equal the commit processed.
+	resp.FlagName = updaterFlag
+	resp.FlagValue = commit
+
+	// Short-circuit if there have been no updates.
+	if commit == dbCommit {
+		log.Debug("no alpine update")
+		return
+	}
+
+	// Append any changed vulnerabilities to the response.
+	for _, namespace := range []string{"v3.3", "v3.4"} {
+		var file io.ReadCloser
+		file, err = os.Open(f.repositoryLocalPath + "/" + namespace + "/main.yaml")
+		if err != nil {
+			return
+		}
+		log.Debug(namespace)
+
+		var vulns []database.Vulnerability
+		switch namespace {
+		case "v3.3":
+			vulns, err = parse33YAML(file)
+		case "v3.4":
+			vulns, err = parse34YAML(file)
+		}
+		if err != nil {
+			return
+		}
+
+		resp.Vulnerabilities = append(resp.Vulnerabilities, vulns...)
+		file.Close()
+	}
+
+	return
+}
+
+func (f *fetcher) pullRepository() (commit string, err error) {
+	if _, pathExists := os.Stat(f.repositoryLocalPath); f.repositoryLocalPath == "" || os.IsNotExist(pathExists) {
+		if f.repositoryLocalPath, err = ioutil.TempDir(os.TempDir(), "alpine-secdb"); err != nil {
+			return "", ErrFilesystem
+		}
+
+		if out, err := utils.Exec(f.repositoryLocalPath, "git", "clone", secdbGitURL, "."); err != nil {
+			f.Clean()
+			log.Errorf("could not pull alpine-secdb repository: %s. output: %s", err, out)
+			return "", cerrors.ErrCouldNotDownload
+		}
+	}
+
+	out, err := utils.Exec(f.repositoryLocalPath, "git", "rev-parse", "HEAD")
+	if err != nil {
+		return "", ErrGitFailure
+	}
+
+	commit = strings.TrimSpace(string(out))
+	return
+}
+
+func (f *fetcher) Clean() {
+	if f.repositoryLocalPath != "" {
+		os.RemoveAll(f.repositoryLocalPath)
+	}
+}
+
+type secdb33File struct {
+	Distro   string `yaml:"distroversion"`
+	Packages []struct {
+		Pkg struct {
+			Name    string   `yaml:"name"`
+			Version string   `yaml:"ver"`
+			Fixes   []string `yaml:"fixes"`
+		} `yaml:"pkg"`
+	} `yaml:"packages"`
+}
+
+func parse33YAML(r io.Reader) (vulns []database.Vulnerability, err error) {
+	var rBytes []byte
+	rBytes, err = ioutil.ReadAll(r)
+	if err != nil {
+		return
+	}
+
+	var file secdb33File
+	err = yaml.Unmarshal(rBytes, &file)
+	if err != nil {
+		return
+	}
+	for _, pack := range file.Packages {
+		pkg := pack.Pkg
+		for _, fix := range pkg.Fixes {
+			version, err := types.NewVersion(pkg.Version)
+			if err != nil {
+				log.Warningf("could not parse package version '%s': %s. skipping", pkg.Version, err.Error())
+				continue
+			}
+
+			var vuln database.Vulnerability
```

Comment:
Is not using `database.Vulnerability{Name: "", Link: ""}` style a conscious decision?